### PR TITLE
add flameshot flatpak manifest

### PR DESCRIPTION
**Pull request by ZetaoYang**
add [flameshot](https://github.com/lupoDharkael/flameshot) flatpak manifest.

### CODE REVIEW

File: org.dharkael.Flameshot.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+app-id: org.dharkael.Flameshot
+runtime: org.kde.Platform
+runtime-version: '5.12'
+sdk: org.kde.Sdk
+command: flameshot
+rename-icon: flameshot
+rename-appdata-file: flameshot.appdata.xml
```
Comment (barthalion) [inline org.dharkael.Flameshot.yaml]: These renames should also happen in upstream repository.

File: org.dharkael.Flameshot.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+rename-desktop-file: flameshot.desktop
+finish-args:
+  - --socket=wayland
+  - --socket=fallback-x11
+  - --socket=x11
```
Comment (barthalion) [inline org.dharkael.Flameshot.yaml]: I'm not exactly sure but `fallback-x11` probably implies `--socket=x11`.

File: org.dharkael.Flameshot.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  - --share=ipc
+  - --device=dri
+  - --filesystem=host # write output files
+  - --filesystem=xdg-config/kdeglobals:ro # gives application access to kdeglobals
+  - --talk-name=com.canonical.AppMenu.Registrar # gives application access to dbus menu
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=org.freedesktop.Notifications
+  - --share=network
+  - --env=PATH=/usr/bin:/app/bin:/app/bin/x86_64-linux
+  - --env=LC_ALL=C
```
Comment (barthalion) [inline org.dharkael.Flameshot.yaml]: Seems odd to enforce C locale?

File: org.dharkael.Flameshot.yaml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+  - --device=dri
+  - --filesystem=host # write output files
+  - --filesystem=xdg-config/kdeglobals:ro # gives application access to kdeglobals
+  - --talk-name=com.canonical.AppMenu.Registrar # gives application access to dbus menu
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=org.freedesktop.Notifications
+  - --share=network
+  - --env=PATH=/usr/bin:/app/bin:/app/bin/x86_64-linux
+  - --env=LC_ALL=C 
+build-options:
```
Comment (barthalion) [inline org.dharkael.Flameshot.yaml]: Such flags are being defined by runtime these days.

File: org.dharkael.Flameshot.yaml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    LD_LIBRARY_PATH: /app/lib:/app/lib64
+
+modules:
+
+- name: flameshot 
+  buildsystem: qmake
+  sources:
+    - type: git
+      url: https://github.com/lupoDharkael/flameshot.git
+      branch: master
```
Comment (barthalion) [inline org.dharkael.Flameshot.yaml]: Use release tag or commit, Flathub targets stable releases.

File: org.dharkael.Flameshot.yaml [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=org.freedesktop.Notifications
+  - --share=network
+  - --env=PATH=/usr/bin:/app/bin:/app/bin/x86_64-linux
+  - --env=LC_ALL=C 
+build-options:
+  cflags: -O2
+  cxxflags: -O2
+  env:
+    LD_LIBRARY_PATH: /app/lib:/app/lib64
```
Comment (barthalion) [inline org.dharkael.Flameshot.yaml]: Build system should be fixed to install files in `/app/lib`.

File: org.dharkael.Flameshot.yaml (hunk 1 above)
Comment (ZetaoYang) [reply org.dharkael.Flameshot.yaml]: removed

File: org.dharkael.Flameshot.yaml (hunk 2 above)
Comment (ZetaoYang) [reply org.dharkael.Flameshot.yaml]: removed `fallback-x11`

File: org.dharkael.Flameshot.yaml (hunk 3 above)
Comment (ZetaoYang) [reply org.dharkael.Flameshot.yaml]: removed

File: org.dharkael.Flameshot.yaml (hunk 4 above)
Comment (ZetaoYang) [reply org.dharkael.Flameshot.yaml]: removed

File: org.dharkael.Flameshot.yaml (hunk 5 above)
Comment (ZetaoYang) [reply org.dharkael.Flameshot.yaml]: add release tag

File: org.dharkael.Flameshot.yaml (hunk 6 above)
Comment (ZetaoYang) [reply org.dharkael.Flameshot.yaml]: removed

**Comment by nedrichards** [comment]
bot, build org.dharkael.Flameshot
testbot, build org.dharkael.Flameshot

**Comment by nedrichards** [comment]
bot, build org.dharkael.Flameshot

**Comment by nedrichards** [comment]
Build fails as it's not correctly compiling the AppStream metadata https://flathub.org/builds/#/builders/32/builds/43 it's worth checking through https://github.com/flathub/flathub/wiki/AppData-Guidelines and making sure that it has all the information and export names that flatpak expects.

**Comment by ZetaoYang** [comment]
@nedrichards added appdata.xml and desktop file.

**Comment by nedrichards** [comment]
bot, build org.dharkael.Flameshot

@ZetaoYang with the latest changes to flathub you should be able to run bot, build commands yourself.

**Comment by nedrichards** [comment]
@ZetaoYang fwiw this appdata file doesn't have all the required elements. https://github.com/flathub/flathub/wiki/App-Requirements#Appstream

**Comment by ZetaoYang** [comment]
bot, build org.dharkael.Flameshot
testbot, build org.dharkael.Flameshot

**Comment by ZetaoYang** [comment]
bot, build org.dharkael.Flameshot

**Comment by nedrichards** [comment]
bot, build org.dharkael.Flameshot

**Comment by Alexander-Wilms** [comment]
You either need to install the appdata file that's part of this PR or add `rename-appdata-file: flameshot.appdata.xml` to the manifest.

**Comment by ZetaoYang** [comment]
bot, build org.dharkael.Flameshot

**Comment by ZetaoYang** [comment]
@Alexander-Wilms Thank you!

**Comment by Alexander-Wilms** [comment]
I tried the test build and noticed that the list of application to open the screenshot with is empty. Maybe this functionality needs to use the xdg-desktop-portal as well?

**Comment by fishxz** [comment]
any news about this?

File: org.dharkael.Flameshot.yaml (hunk 4 above)
Comment (barthalion) [reply org.dharkael.Flameshot.yaml]: I think you forgot to push it.

File: org.dharkael.Flameshot.yaml [hunk 7]
```diff
@@ -0,0 +1,31 @@
+app-id: org.dharkael.Flameshot
+runtime: org.kde.Platform
+runtime-version: '5.12'
+sdk: org.kde.Sdk
+command: flameshot
+rename-icon: flameshot  
+rename-appdata-file: flameshot.appdata.xml
```
Comment (barthalion) [inline org.dharkael.Flameshot.yaml]: These renames should happen upstream too.

File: org.dharkael.Flameshot.desktop [hunk 8]
```diff
@@ -0,0 +1,48 @@
+[Desktop Entry]
```
Comment (barthalion) [inline org.dharkael.Flameshot.desktop]: You are not using the file in the manifest, so I assume upstream's one is used. Please contribute your changes there, if any.

File: org.dharkael.Flameshot.appdata.xml [hunk 9]
```diff
@@ -0,0 +1,65 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline org.dharkael.Flameshot.appdata.xml]: You are not using the file in the manifest, so I assume upstream's one is used. Please contribute your changes there, if any.

**Pull request closed without merging**